### DISCUSSION TITLE
How to use proxies (and which service to use)?

### DISCUSSION THREAD

**Discussion by davidtgq** [jef/streetmerchant#2527] [Help]
I'd like to set up proxy to work around rate limits. How is this supposed to work, and what proxy service provider do people use with streetmerchant?

I've tried this in global.proxies, but only get the message `Navigation timeout of 30000 ms exceeded`:
```
socks5://H53ud:GsX2B7@amsterdam.nl.socks.nordhold.net
socks5://H53ud:GsX2B7@atlanta.us.socks.nordhold.net
socks5://H53ud:GsX2B7@dallas.us.socks.nordhold.net
socks5://H53ud:GsX2B7@dublin.ie.socks.nordhold.net
socks5://H53ud:GsX2B7@ie.socks.nordhold.net
socks5://H53ud:GsX2B7@los-angeles.us.socks.nordhold.net
socks5://H53ud:GsX2B7@nl.socks.nordhold.net
socks5://H53ud:GsX2B7@se.socks.nordhold.net
socks5://H53ud:GsX2B7@stockholm.se.socks.nordhold.net
socks5://H53ud:GsX2B7@us.socks.nordhold.net
```

I also tried this, but only get the message `net::ERR_FAILED`:
```
http://47.116.76.219:80
http://47.244.50.194:8081
http://67.43.234.114:3128
http://137.74.112.21:80
http://47.241.16.190:80
http://142.116.32.230:80
http://47.57.188.208:80
https://159.89.221.73:3128
```

**Comment by KOLBASOID** [comment]
socks5://H53ud:GsX2B7@IP:PORT

**Comment by jiglesiasc** [comment]
I have the proxy working well for me with the global.proxies file. Just do as this: protocol://[user:password@]ip[:port]

As a proxy server I use privateproxy.me. Is pretty fast and charges $5/month/IP. Is quite fast

**Comment by V3rgE** [comment]
If I'm using freeproxylists.net for my PROXY_ADDRESS how do I format it? PROXY_ADDRESS=http://IP:PORT   doesn't get me anywhere (Target Closed error). And when I tried using that "socks5://H53ud:GsX2B7@IP:PORT" that @KOLBASOID suggested it didn't work either (although it only says "ERR_NO_SUPPORTED_PROXIES" with this. How am I supposed to do this? I have PROXY_PROTOCOL=http    and  PROXY_PORT=80
What am I missing?

**Reply by WeDoSys** [reply]
@davidtgq 
just FYI i think the entire world is using your proxies right now... use "user" "pass" when commenting your user and password so your details aren't revealed to everyone. 
I also was getting timeout errors until i upgraded to a better proxy provider, shared proxies seem to be less reliable.
Good Luck
